docker/Dockerfile

```diff
@@ -5,12 +5,14 @@ RUN apt-get update && apt-get install -y -q \
     libmpich-dev 
 
 ADD ./requirements.txt /app/requirements.txt
+ADD ./requirements-optional.txt /app/requirements-optional.txt
```

Comment:
maybe could be added as an optional build arguments. You can look at `Dockerfile.pipeline` to see how to mange it and add the if statement, could extend to the libmpich above if not needed

Reply:
I think it's better to include all optional dependencies needed to run all tutorials as this is proving to be a big issue at every hackathon and workshop we participate in - even with otherwise experienced users. 'Experts' can always create whatever bespoke Dockerfile they need as by definition they 'know what they are doing'.